text/entities/0256-entities-data-model.md

```diff
@@ -0,0 +1,790 @@
+- [Entity Identification](#entity-identification)
+  * [LID, GID and IDCONTEXT](#lid-gid-and-idcontext)
+  * [Semantic Conventions](#semantic-conventions)
+  * [Examples](#examples)
+    + [Process in a Host](#process-in-a-host)
+    + [Process in Kubernetes](#process-in-kubernetes)
+    + [Host in Cloud Account](#host-in-cloud-account)
+- [Prototypes](#prototypes)
+- [Prior Art](#prior-art)
+- [Alternatives](#alternatives)
+  * [Different ID Structure](#different-id-structure)
+  * [No Entity Events](#no-entity-events)
+  * [Merge Entity Events data into Resource](#merge-entity-events-data-into-resource)
+  * [Hierarchical ID Field](#hierarchical-id-field)
+- [Open questions](#open-questions)
+  * [Attribute Data Type](#attribute-data-type)
+  * [Classes of Entity Types](#classes-of-entity-types)
+  * [Multiple Observers](#multiple-observers)
+  * [Is Type part of Entity's identity?](#is-type-part-of-entitys-identity)
+  * [Choosing from Multiple Ids](#choosing-from-multiple-ids)
+- [Future Work](#future-work)
+- [References](#references)
+
+<!-- tocstop -->
+
+## Motivation
+
+This data model sets the foundation for adding entities to OpenTelemetry. The data model
+is largely borrowed from
+[the initial proposal](https://docs.google.com/document/d/1VUdBRInLEhO_0ABAoiLEssB1CQO_IcD5zDnaMEha42w/edit)
+that was accepted for entities SIG formation.
+
+This OTEP is step 1 in introducing the entities data model. Follow up OTEPs will add
+further data model definitions, including the linking of Resource information
+to entities.
+
+## Design Principles
+
+- Consistency with the rest of OpenTelemetry is important. We heavily favor solutions
+  that look and feel like other OpenTelemetry data models.
+
+- Meaningful (especially human-readable) IDs are more valuable than random-generated IDs.
+  Long-lived IDs that survive state changes (e.g. entity restarts) are more valuable than
+  short-lived, ephemeral IDs.
+  See [the need for navigation](https://docs.google.com/document/d/1Xd1JP7eNhRpdz1RIBLeA1_4UYPRJaouloAYqldCeNSc/edit#heading=h.fut2c2pec5wa).
+
+- We cannot make an assumption that the entirety of information that is necessary for
+  global identification of an entity is available at once, in one place. This knowledge
+  may be distributed across multiple participants and needs to be combined to form a
+  globally unique identifier.
+
+- Semantic conventions must bring as much order as possible to telemetry, however they
+  cannot be too rigid and prevent real-world use cases.
+
+## Data Model
+
+We propose a new concept of Entity.
+
+Entity represents an object of interest associated with produced telemetry:
+traces, metrics or logs.
+
+The Entity that produces the telemetry is called the Producing Entity. For example
+telemetry produced using OpenTelemetry SDK is normally associated with a Service.
+The Service is the Producing Entity. Similarly, OpenTelemetry defines system metrics
+for a host. The Host is the Producing Entity in this case.
```

Comment:
I find the term "Producing" to be confusing here because although the host is described by the telemetry, it is not necessarily _producing_ it. For example, an external observer may be generating the telemetry based on observations of the host, in which case it seems more natural to describe the external observer as the producer.

To highlight this further, I believe line 440 uses the term "producer" in the "external observer" sense rather than the "described entity" sense: `The entity information producer is responsible for determining the identification context of each entity it is producing information about.`

If I understand the intention here, we're really just trying to be clear that the Entity is the object that is described by the telemetry, so would it be clearer to call it the "Described Entity". Moreover, is either adjective really adding value here? Why not just use the term Entity?

Reply:
This is a good call out.  In the previous "full" proposal, there's a link between "Resource" and "Producing Entity".  I think the language you see here is a bit of a preview of a Resource<->Entity mapping.   I think it'd be better to cut this from the proposal (just use entity) prior to needing the term "Producing Entity" and associations between Resource<->Entity, which should be the next OTEP we read/agree on.

Reply:
I agree. We arlready have a TODO to find a better name: https://github.com/orgs/open-telemetry/projects/85/views/1?pane=issue&itemId=57054361

I removed references to "Producing Entity" from this OTEP for now in commit https://github.com/open-telemetry/oteps/pull/256/commits/a1f126f339a9078235819371af24b2a31153a968.

We will find a better name in follow up OTEPs.